lib/holdings/item.rb

```diff
@@ -154,6 +154,34 @@ def request_link
       @request_link ||= ItemRequestLinkComponent.new(item: self)
     end
 
+    def current_location_is_always_requestable?
+      return true if current_location.code.end_with?('-LOAN') && current_location.code != "SEE-LOAN"
+
+      (Settings.requestable_current_locations[library] || Settings.requestable_current_locations.default).include?(current_location.code) ||
+        (Settings.unavailable_current_locations[library] || Settings.unavailable_current_locations.default).include?(current_location.code)
+    end
+
+    def in_mediated_pageable_location?
+      Settings.mediated_locations[library] == '*' ||
+        Settings.mediated_locations[library]&.include?(home_location)
+    end
+
+    def aeon_pageable?
+      Settings.aeon_locations[library] == '*' ||
+        Settings.aeon_locations.dig(library, home_location) == "*" ||
+        Settings.aeon_locations.dig(library, home_location)&.include?(type)
+    end
+
+    def in_nonrequestable_location?
+      (Settings.nonrequestable_locations[library] || Settings.nonrequestable_locations.default).include?(home_location)
+    end
+
+    def requestable?
```

Comment:
If we're going to do this refactor, maybe `#requestable?` isn't a good name for the method. We're certainly confusing item properties (like a non-requestable location or being on reserve), but an item is still requestable aeon-paged or mediated-paged, we just don't show an item-level link for it because the location-level link takes care of it.

Reply:
Alright, I had another stab at this, extracting the business logic about requests link display to a separate class.